app/shared/components/Tools/Form/CreateAccount.js

```diff
@@ -0,0 +1,420 @@
+// @flow
+import React, { Component } from 'react';
+import { translate } from 'react-i18next';
+import { Decimal } from 'decimal.js';
+
+import { Segment, Form, Divider, Button, Message } from 'semantic-ui-react';
+
+import GlobalFormFieldRam from '../../Global/Form/Field/Ram';
+import GlobalFormFieldToken from '../../Global/Form/Field/Token';
+import GlobalFormFieldAccount from '../../Global/Form/Field/Account';
+import GlobalFormFieldKeyPublic from '../../Global/Form/Field/Key/Public';
+import FormMessageError from '../../Global/Form/Message/Error';
+import ToolsFormCreateAccountConfirming from './CreateAccount/Confirming';
+import calculatePriceOfRam from '../../helpers/calculatePriceOfRam';
+
+type Props = {
+  actions: {},
+  globals: {},
+  system: {}
+};
+
+const formAttributes = ['accountName', 'activeKey', 'ownerKey', 'delegatedBw', 'delegatedCpu', 'ramAmount'];
+
+class ToolsFormCreateAccount extends Component<Props> {
+  props: Props;
+
+  constructor(props) {
+    super(props);
+
+    const {
+      balance
+    } = props;
+
+    this.state = {
+      confirming: false,
+      EOSbalance: (balance && balance.EOS) ? balance.EOS : 0,
+      formErrors: {},
+      submitDisabled: true
+    };
+  }
+
+  componentDidMount() {
+    this.tick();
+    this.interval = setInterval(this.tick.bind(this), 30000);
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.interval);
+  }
+
+  tick() {
+    const {
+      actions,
+    } = this.props;
+
+    const {
+      getRamStats
+    } = actions;
+
+    getRamStats();
+  }
+
+  onSubmit = (e) => {
+    if (!this.state.submitDisabled) {
+      this.setState({
+        confirming: true
+      });
+    }
+    e.preventDefault();
+    return false;
+  }
+
+  onKeyPress = (e) => {
+    if (e.key === 'Enter') {
+      this.onSubmit(e);
+
+      e.preventDefault();
+      return false;
+    }
+  }
+
+  onChange = (e, { name, value, valid }) => {
+    this.setState({
+      submitDisabled: false,
+      [name]: value
+    }, () => {
+      let {
+        formErrors,
+        ramPrice
+      } = this.state;
+
+      const {
+        accountName
+      } = this.state;
+
+      const {
+        actions,
+        globals,
+        system
+      } = this.props;
+
+      const {
+        checkAccountAvailability
+      } = actions;
+
+      if (name === 'accountName' &&
+          accountName.length === 12 &&
```

Comment:
Didn't we talk about not requiring 12 characters?

Reply:
This is not requiring it, but rather displaying a warning saying that only characters that have been bid on can have less than 12 characters.

Reply:
sorry this is actually only checking account availability for accounts that have 12 characters, since most users are going to use account names with 12 characters, I figured that it would be better not to the account name lookup until all 12 characters are entered. I'm down for whatever you think is best though 😄

Reply:
Yeah, I'd remove that. The minute someone tries to create a non-12 character name, the lookup isn't going to work.

Reply:
👍